File: XMLCheck/Handler/CheckRulesXML14/Xml14RuleRunner.cs
using XMLCheck.Handler.CheckRulesXML13;
using XMLCheck.Models;
using XMLCheck.Services;

namespace XMLCheck.Handler.CheckRulesXML14
{
    public static class Xml14RuleRunner
    {
        private static readonly List<IXml14Rule> _rules = new()
        {

        };
        public static List<ErrorDetails> Run(Xml14Model? model,HoSoContext context)
        {
            var errors = new List<ErrorDetails>();

            foreach (var rule in _rules)
            {
                var err = rule.Check(model,context);
                if (err != null)
                    errors.Add(err);
            }

            return errors;
        }
    }
}

File: XMLValidator_v2/src/parsers/rules/xml14/xml14RuleRunner.ts
import { Xml14Model } from '../../../interfaces/xml-models.interface';
import { HoSoContext } from '../../../services/hoso-context';
import { ErrorDetails } from '../../../dto/xml-check.dto';
import { IXml14Rule } from './xml14RuleBase';

export class Xml14RuleRunner {
  private static readonly rules: IXml14Rule[] = [];

  public static run(model: Xml14Model, context: HoSoContext): ErrorDetails[] {
    const errors: ErrorDetails[] = [];
    if (!model) return errors;

    for (const rule of this.rules) {
      const err = rule.check(model, context);
      if (err) {
        errors.push(err);
      }
    }

    return errors;
  }
}
